lms/djangoapps/badges/backends/badgr.py

```diff
@@ -166,12 +179,102 @@ def _create_assertion(self, badge_class, user, evidence_url):
         self._send_assertion_created_event(user, assertion)
         return assertion
 
-    @staticmethod
-    def _get_headers():
+    def _encrypt_token(self, token):
+        """
+        Use the SECRET_KEY to encrypt a token.
+        """
+        fernet_key = base64.urlsafe_b64encode(
+            settings.SECRET_KEY.ljust(64).encode('utf-8')[:32]
+        )
+        f = Fernet(fernet_key)
```

Comment:
You might still want to factor this bit out into a helper method, which you call from here and from `_decrypt_token()` (DRY).

Reply:
Added a helper method.